shuffle/cli/src/deploy.rs

```diff
@@ -86,34 +99,45 @@ pub fn send_module_transaction(
                 factory.payload(TransactionPayload::Module(Module::new(binary))),
             );
 
-            send(client, publish_txn)?;
+            let bytes = bcs::to_bytes(&publish_txn)?;
+            let hash = send_txn(bytes, network)?;
+            vec.push(hash);
         } else {
             println!("Skipping Module: {}", module_id);
         }
     }
     println!("Success!");
-    Ok(())
+    Ok(vec)
 }
 
-pub fn check_module_exists(client: &BlockingClient, account: &LocalAccount) -> Result<()> {
-    let account_state_blob: AccountStateBlob = {
-        let blob = client
-            .get_account_state_with_proof(account.address(), None, None)?
-            .into_inner()
-            .blob
-            .ok_or_else(|| anyhow::anyhow!("missing account state blob"))?;
-        bcs::from_bytes(&blob)?
-    };
-    let account_state = AccountState::try_from(&account_state_blob)?;
-    let mut modules = vec![];
-    for module_bytes in account_state.get_modules() {
-        modules.push(normalized::Module::new(
-            &CompiledModule::deserialize(module_bytes)
-                .map_err(|e| anyhow!("Failure deserializing module: {:?}", e))?,
-        ));
+pub fn send_txn(txn_bytes: Vec<u8>, network: &str) -> Result<String> {
+    let path = Url::parse(format!("http://{}/transactions", network).as_str())?;
+    let resp = ureq::post(path.as_str())
+        .set("Content-Type", "application/vnd.bcs+signed_transaction")
+        .send_bytes(&*txn_bytes);
+    let json: serde_json::Value = resp.into_json()?;
+    let hash = json["hash"].as_str().unwrap();
+    Ok(hash.to_string())
+}
+
+pub fn confirm_txn_executed(hash: &str, network: &str) -> Result<()> {
+    let path = Url::parse(format!("http://{}/transactions/{}", network, hash).as_str())?;
```

Comment:
Same problem here, should not hard code http

Reply:
Discussed offline but I'm removing hard coded protocols. Now user passes in full Url